### DISCUSSION TITLE
Calculation of td_log in heat exchangers with constant temperature difference

### DISCUSSION THREAD

**Discussion by dk-teknologisk-enp** [oemof/tespy#394] [Ideas]
Currently it seems that if ttd_u == ttd_l for a heat exchanger, then td_log ends up as nan, and thus also the kA.

However, this is a perfectly valid situation, for example in a heat exchanger with purely boiling/condensing on both sides.

I suggest making the following changes to base.py, such that td_log is set to the constant temperature difference in case ttd_u == ttd_l:

![image](https://user-images.githubusercontent.com/124161512/221202216-e5fe14ef-0504-41ac-a52f-0143149971ad.png)
(base.py:838-840)

**Comment by fwitte** [comment]
Dear @dk-teknologisk-enp,

that sounds like a very good idea. It seems I have never come across this bug in 6 years of development and using the software. Thanks a lot. Do you like to make the changes in a pull request yourself? I‘ll be in holidays one more week and can only implement the changes once I get back to work.

best

Francesco

**Comment by dk-teknologisk-enp** [comment]
Hi Francesco,

I have made this PR: https://github.com/oemof/tespy/pull/396 

I have not added any tests, but that might be a good idea.
